catalog/app/containers/Admin/Sync.tsx

```diff
@@ -0,0 +1,306 @@
+
+interface DataRow {
+  id?: string
+  local: string
+  s3: string
+}
+
+function useSyncFolders(): [null | DataRow[], () => void] {
+  const ipc = IPC.use()
+  const [key, setKey] = React.useState(1)
+  const inc = React.useCallback(() => setKey(R.inc), [setKey])
+  const [folders, setFolders] = React.useState<null | DataRow[]>(null)
+  React.useEffect(() => {
+    async function fetchData() {
+      const config = await ipc.invoke(IPC.EVENTS.SYNC_FOLDERS_LIST)
+      setFolders(config.folders)
+    }
+
+    fetchData()
+  }, [ipc, key])
+  return folders ? [folders, inc] : [null, inc]
+}
+
+interface ConfirmDeletionDialogProps {
+  onCancel: () => void
+  onSubmit: (v: DataRow) => void
+  value: DataRow | null
+}
+
+function ConfirmDeletionDialog({
+  onCancel,
+  onSubmit,
+  value,
+}: ConfirmDeletionDialogProps) {
+  const handleSubmit = React.useCallback(() => {
+    if (value) onSubmit(value)
+  }, [onSubmit, value])
+  return (
+    <M.Dialog open={!!value}>
+      <M.DialogTitle>Remove local ⇄ s3 folder pair</M.DialogTitle>
+      <M.DialogContent>
+        Confirm deletion of {value?.local}⇄{value?.s3} sync pair
+      </M.DialogContent>
+      <M.DialogActions>
+        <M.Button onClick={onCancel} color="primary">
+          Cancel
+        </M.Button>
+        <M.Button color="primary" onClick={handleSubmit} variant="contained">
+          Remove
+        </M.Button>
+      </M.DialogActions>
+    </M.Dialog>
+  )
+}
+
+interface ManageFolderDialogProps {
+  onCancel: () => void
+  onSubmit: (v: DataRow) => void
+  value: Partial<DataRow> | null
+}
+
+function ManageFolderDialog({ onCancel, onSubmit, value }: ManageFolderDialogProps) {
+  return (
+    <M.Dialog open={!!value}>
+      <RF.Form onSubmit={onSubmit} initialValues={value}>
+        {({ handleSubmit, submitting, submitFailed, hasValidationErrors }) => (
+          <>
+            <M.DialogTitle>Add local ⇄ s3 folder pair</M.DialogTitle>
+            <M.DialogContent>
+              <RF.Field
+                component={LocalFolderInput}
+                name="local"
+                label="Local folder"
+                placeholder="Folder on local file system"
+                validate={validators.required as FF.FieldValidator<any>}
+                errors={{
+                  required: 'Path to local directory is required',
+                }}
+                fullWidth
+                margin="normal"
+              />
+              <RF.Field
+                component={Form.Field}
+                name="s3"
+                label="S3 bucket + Package name"
+                placeholder="s3://bucket/namespace/package"
+                validate={
+                  validators.composeAnd(
+                    validators.required,
+                    validators.s3Url,
+                  ) as FF.FieldValidator<any>
+                }
+                errors={{
+                  required: 'S3 URL is required',
+                  s3Url: 'Enter valid S3 url to package',
```

Comment:
```suggestion
                  s3Url: 'Enter valid S3 URL to package',
```
this is not technically correct, but whatever, it makes sense in this context